src/flows/context/query.tsx

```diff
@@ -81,11 +96,14 @@ const findOrgID = (text, buckets) => {
 
 export const QueryProvider: FC = ({children}) => {
   const {flow} = useContext(FlowContext)
+  const {runMode} = useContext(RunModeContext)
+  const {add, update} = useContext(ResultsContext)
   const variables = useSelector((state: AppState) => getVariables(state))
   const buckets = useSelector((state: AppState) => getSortedBuckets(state))
   const bucketsLoadingState = useSelector((state: AppState) =>
     getStatus(state, ResourceType.Buckets)
   )
+  const [isLoading, setLoading] = useState(RemoteDataState.NotStarted)
```

Comment:
this isn't it's own state, but a derived state. it's not that `queryAll` sets the state and keeps track of it, but that it's a conglomeration of all of the stages loading state. instead of a useState, it's more of a `flows.meta.all.reduce(meta => if any are loading, show the loading state), RemoteDataState.NotStarted)'

Reply:
and yes i know that's not how it worked before. but i think we're in a better place now with this

Reply:
Gotcha. SO then this should be obtainable in the submit context